internal/gps/vcs_version.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gps
+
+import (
+	"strings"
+
+	"github.com/Masterminds/vcs"
+	"github.com/pkg/errors"
+)
+
+// ProjectVersion returns the current project version for an absolute path.
+func ProjectVersion(path string) (Version, error) {
```

Comment:
So I'm generally OK with having this logic in gps, except that I think we should avoid suggesting that this is getting versions for projects. There's a not-implausible future in which it's possible that we'd allow versions to be a property of metadata within the file tree itself (akin to `Cargo.toml`). In that case, it'd be up to the tool to impose these semantics.

Instead, let's use a weaker, more functionally-accurate name for the func: `VCSVersion`